packages/service-library/src/servicelib/rabbitmq.py

```diff
@@ -259,8 +268,8 @@ async def publish(
                 durable=True,
             )
             await exchange.publish(
-                aio_pika.Message(message.encode()),
-                routing_key=topic or "",
+                aio_pika.Message(message.body()),
+                routing_key=message.routing_key() or "",
```

Comment:
it depends on the exchange type.
If FANOUT: routing key does nothing. e.g. you can put there whatever you want, it will be transmitted anyway.
if DIRECT: message with routing_key == queue.binding.routing_key will be routed to the queue, others are discarded.
if TOPIC: message with routing_key in queue.binding.routing_key (including special binding such as ```#``` or ```*``` will be used. if nothing is found the message is discarded.